include/deal.II/base/array_view.h

```diff
@@ -252,35 +259,97 @@ class ArrayView
    */
   const std::size_t n_elements;
 
-  friend class ArrayView<const ElementType>;
+  friend class ArrayView<const ElementType, MemorySpaceType>;
 };
 
 
 
 //---------------------------------------------------------------------------
 
 
+namespace internal
+{
+  namespace ArrayViewHelper
+  {
+    template <typename MemorySpaceType>
+    inline bool
+    correct_memory_space(const void *const)
```

Comment:
I usually take the first word of a function name as a verb, which would here suggest that you are correct*ing* something. But I think you really mean to check that. So maybe `is_correct_memory_space()`?